testing-common/integration-tests/src/main/java/indy/IndyInstrumentationTestModule.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package indy;
+
+import static net.bytebuddy.implementation.bytecode.assign.Assigner.Typing.DYNAMIC;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import java.util.Collections;
+import java.util.List;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.asm.Advice.AssignReturned.ToArguments.ToArgument;
+import net.bytebuddy.asm.Advice.AssignReturned.ToFields.ToField;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+@AutoService(InstrumentationModule.class)
+public class IndyInstrumentationTestModule extends InstrumentationModule {
+
+  public IndyInstrumentationTestModule() {
+    super("indy-test");
+  }
+
+  @Override
+  public boolean isIndyModule() {
+    return true;
+  }
+
+  @Override
+  public List<TypeInstrumentation> typeInstrumentations() {
+    return Collections.singletonList(new Instrumentation());
+  }
+
+  @Override
+  public boolean isHelperClass(String className) {
+    return className.equals(LocalHelper.class.getName());
+  }
+
+  public static class Instrumentation implements TypeInstrumentation {
+
+    @Override
+    public ElementMatcher<TypeDescription> typeMatcher() {
+      return named("indy.IndyInstrumentationTest");
+    }
+
+    @Override
+    public void transform(TypeTransformer transformer) {
+      String prefix = getClass().getName();
+      transformer.applyAdviceToMethod(
+          named("assignToFieldViaReturn"), prefix + "$AssignFieldViaReturnAdvice");
+      transformer.applyAdviceToMethod(
+          named("assignToFieldViaArray"), prefix + "$AssignFieldViaArrayAdvice");
+      transformer.applyAdviceToMethod(
+          named("assignToArgumentViaReturn"), prefix + "$AssignArgumentViaReturnAdvice");
+      transformer.applyAdviceToMethod(
+          named("assignToArgumentViaArray"), prefix + "$AssignArgumentViaArrayAdvice");
+      transformer.applyAdviceToMethod(
+          named("assignToReturnViaReturn"), prefix + "$AssignReturnViaReturnAdvice");
+      transformer.applyAdviceToMethod(
+          named("assignToReturnViaArray"), prefix + "$AssignReturnViaArrayAdvice");
+      transformer.applyAdviceToMethod(named("getHelperClass"), prefix + "$GetHelperClassAdvice");
+      transformer.applyAdviceToMethod(named("exceptionPlease"), prefix + "$ThrowExceptionAdvice");
+      transformer.applyAdviceToMethod(
+          named("noExceptionPlease"), prefix + "$SuppressExceptionAdvice");
+    }
+
+    @SuppressWarnings({"unused", "CanIgnoreReturnValueSuggester"})
```

Comment:
Nit: perhaps we could think of ignoring `CanIgnoreReturnValueSuggester` for classes that end with `...Advice`; we're already doing a similar thing for the `PrivateConstructorForUtilityClass` check

See https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/f89fa74852b3ffa25facb6e6d6c67cbd7bfb4328/conventions/src/main/kotlin/otel.errorprone-conventions.gradle.kts#L128-L131 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/custom-checks/src/main/java/io/opentelemetry/javaagent/customchecks/OtelPrivateConstructorForUtilityClass.java

Reply:
I've implemented an Otel variant of the check now. It is slightly more compex than the `OtelPrivateConstructorForUtilityClass` because it is a method-level check. Therefore I need to find the declaring class manually in the check.

If you know any better solution here let me know.